Engine/TransactionHandlers/BrokerageTransactionHandler.cs

```diff
@@ -609,6 +615,11 @@ private OrderResponse HandleSubmitOrderRequest(SubmitOrderRequest request)
             // rounds off the order towards 0 to the nearest multiple of lot size
             order.Quantity = RoundOffOrder(order, security);
 
+            if (order.Quantity == 0)
+            {
+                return OrderResponse.ZeroQuantity(request);
```

Comment:
Regardless of where the `RoundOffOrder` behavior ends up, I think this check should be here. It seems very reasonable that a transaction handler would prevent zero quantity orders.